### ISSUE TITLE
Implement job restart policies

### ISSUE THREAD

**Issue by razvan** [stackabletech/spark-k8s-operator#157]
### Description

Some Spark jobs are idempotent and can be resubmitted without problems especially if they fail due to external constraints (like endpoints not being ready and such)

The user should be able to configure such jobs with a restart policy.

**Comment by sbernauer** [comment]
At my previous company i simply put the Spark driver Pod within a Deployment and health-checked port 4040.
Worked pretty well

**Comment by sbernauer** [comment]
Yes, it is. I mentioned it for the sake of completeness, and our approach is likely better :)
